### DISCUSSION TITLE
Image lazy loading - rootMargin

### DISCUSSION THREAD

**Discussion by jdulik** [vercel/next.js#24552] [Help]
Is it any equivalent in the next.js Image component to [rootMargin](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/rootMargin) in the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)?

**Comment by miguderp** [comment]
Curious as well to learn if there are some solutions here, or else I'll have my designers force me to stop using `next/image`…

**Reply by jdulik** [reply]
I have already solved this problem. I used [react-intersection-observer library](https://www.npmjs.com/package/react-intersection-observer) and depending on whether image is in the view I load it using "priority" prop.

[Full solution](https://gist.github.com/dulikkk/2a2102aec4ed901dac6db66f6f606f3c)

**Reply by miguderp** [reply]
Thanks for sharing!

**Reply by vinaynair39** [reply]
For some reason, the root margin is not working for me, Any idea what might be the issue?

**Reply by Ryz0nd** [reply]
Use the lazyBoundary prop! [Next 11.0.2-canary.21](https://github.com/vercel/next.js/releases/tag/v11.0.2-canary.21)

**UPDATE**
See: [Next 11.1.0](https://github.com/vercel/next.js/releases/tag/v11.1.0)

**Reply by vinaynair39** [reply]
@Ryz0nd thank you so much man. You're a life saver. Any idea when this will be released?

**Reply by styfle** [reply]
You can try it out today with `yarn add next@canary`. It will be available in the next stable release sometime in the coming weeks.

**Comment by tslater** [comment]
It looks like lazyBoundary and lazyRoot were removed from the newer Image component. Is there a way to calibrate the loading? Is Image now just using the native tag? Not sure how to get things to load just a bit sooner without using legacy.

**Reply by styfle** [reply]
- `next/image` uses native `loading="lazy"` so there is no way to configure it - the browser decides the best time to load the image using different heuristics
- `next/legacy/image` uses IntersectionObserver which [can be configured](https://nextjs.org/docs/pages/api-reference/components/image-legacy#lazyboundary) but it will typically be slower because it requires javascript to load prior to calculating position and then loading the image
